azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -681,6 +692,8 @@ func resourceKubernetesClusterNodePoolRead(d *schema.ResourceData, meta interfac
 
 		d.Set("vnet_subnet_id", props.VnetSubnetID)
 		d.Set("vm_size", string(props.VMSize))
+
+		d.Set("upgrade_settings", flattenUpgradeSettings(props.UpgradeSettings))
```

Comment:
(as above) since this is a complex object this wants to be:

```suggestion
		if err := d.Set("upgrade_settings", flattenUpgradeSettings(props.UpgradeSettings)); err != nil {
			return fmt.Errorf("setting `upgrade_settings`: %+v", err)
		}
```